servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -240,11 +249,56 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 builder.sslConfig(DEFAULT_CLIENT_SSL_CONFIG);
             }
 
+            MultiAddressStrategyWrapper wrapper = new MultiAddressStrategyWrapper();
+            builder.appendClientFilter(wrapper);
+
             if (singleAddressInitializer != null) {
                 singleAddressInitializer.initialize(urlKey.scheme, urlKey.hostAndPort, builder);
             }
 
-            return builder.buildStreaming();
+            StreamingHttpClient singleClient = builder.buildStreaming();
+
+            wrapper.clientStrategy = singleClient.executionContext().executionStrategy();
+            return singleClient;
+        }
+    }
+
+    private static final class MultiAddressStrategyWrapper implements StreamingHttpClientFilterFactory {
+
+        HttpExecutionStrategy clientStrategy = defaultStrategy();
+        @Override
+        public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+            return new StreamingHttpClientFilter(client) {
+                @Override
+                protected Single<StreamingHttpResponse> request(
+                        final StreamingHttpRequester delegate, final StreamingHttpRequest request) {
+                    applyMultiAddressStrategy(request.context(), clientStrategy);
+                    return super.request(delegate, request);
+                }
+
+                private HttpExecutionStrategy applyMultiAddressStrategy(
+                        final ContextMap contextMap, final HttpExecutionStrategy singleClientStrategy) {
+                    HttpExecutionStrategy requestStrategy =
+                            contextMap.getOrDefault(HTTP_EXECUTION_STRATEGY_KEY, defaultStrategy());
+                    HttpExecutionStrategy useStrategy = defaultStrategy() == requestStrategy ?
+                            offloadAll() :
+                            defaultStrategy() == singleClientStrategy || !singleClientStrategy.hasOffloads() ?
+                                    // single client is default or has no *additional* offloads
+                                    requestStrategy :
+                                    // add single client offloads to existing strategy
+                                    requestStrategy.merge(singleClientStrategy);
```

Comment:
Intellij IDEA suspects NPE here. Consider adding `assert requestStrategy != null;` after you take `requestStrategy` from the `contextMap`